From b032955dfac213c79a726931bcd4770bae817a24 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Thu, 4 Jul 2024 16:11:52 +0300
Subject: [PATCH] libtool: Alter expected line endings in testsuite

When comparing expected output in LT_AT_CHECK_EXECUTE, several tests
fail with mismatched line endings. This patch should fix Cygwin
environments that have CRLF text line endings disabled and only use
the LF character for line endings.

* tests/testsuite.at: Removed '\r' from awk's printf command to support
  Cygwin in testsuite.
---
 tests/testsuite.at | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/tests/testsuite.at b/tests/testsuite.at
index 093cf03c..1008102b 100644
--- a/tests/testsuite.at
+++ b/tests/testsuite.at
@@ -273,8 +273,14 @@ AT_CHECK([test -z "$leftovers"])
 # Ensure correct line ending for expected output of host executables.
 m4_define([LT_AT_HOST_DATA],
 [AT_DATA([$1], [$2])
-case $host_os in mingw* | windows*)
-  awk '{printf ("%s\r\n", [$]0);}' < $1 > $1.t && mv -f $1.t $1 ;;
+case $host_os in
+mingw* | windows*)
+  awk '{printf ("%s\r\n", [$]0);}' < $1 > $1.t && mv -f $1.t $1
+  ;;
+
+cygwin*)
+  awk '{printf ("%s\n", [$]0);}' < $1 > $1.t && mv -f $1.t $1
+  ;;
 esac])
 
 
-- 
2.45.1

